src/main/java/picard/sam/DownsampleSam.java

```diff
@@ -212,7 +218,28 @@ protected int doWork() {
         }
 
         final SamReader in = SamReaderFactory.makeDefault().referenceSequence(REFERENCE_SEQUENCE).open(SamInputResource.of(INPUT));
-        final SAMFileWriter out = new SAMFileWriterFactory().makeSAMOrBAMWriter(in.getFileHeader(), true, OUTPUT);
+        final SAMFileHeader header = in.getFileHeader().clone();
+
+        if (STRATEGY == Strategy.ConstantMemory || STRATEGY == Strategy.Chained) {
+            //if running using ConstantMemory or Chained strategy, need to check if we have previously run with the same random seed
+            //collect previously used seeds
+            final Set<Integer> previousSeeds = new HashSet<>();
+            for (final SAMProgramRecord pg : header.getProgramRecords()) {
+                if (pg.getProgramName() != null && pg.getProgramName().equals(PG_PROGRAM_NAME)) {
+                    final int previousSeed = Integer.parseInt(pg.getAttribute(RANDOM_SEED_TAG));
+                    previousSeeds.add(previousSeed);
+                }
+            }
+
+            while (previousSeeds.contains(RANDOM_SEED)) {
+                RANDOM_SEED *= 2;
```

Comment:
This is too small a space to traverse. I would pull out random seed from a RNG that itself is initialized with the provided seed.

Reply:
also, if you modified the seed, put out a warning that the provided random seed had already been used and that you are using a different seed (and state what it is)

Reply:
done